src/Microsoft.AspNetCore.Routing.Abstractions/IEndpointFeature.cs

```diff
@@ -0,0 +1,17 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNetCore.Http;
+
+namespace Microsoft.AspNetCore.Routing
+{
+    public interface IEndpointFeature
+    {
+        Endpoint Endpoint { get; set; }
+
+        Func<RequestDelegate, RequestDelegate> Invoker { get; set; }
+
+        RouteValueDictionary Values { get; set; }
```

Comment:
@davidfowl do you have thoughts on whether you want `RVD` to move to the HTTP repo or should we change  this to be `IDictionary<string, object>`?

Reply:
We need to file an issue for this and have that discussion. We should be crisp about what the core exchange types are here. I would be fine moving RVD to HttpAbstractions. We need to rename the other 2 types but those look reasonable as well.